### PR TITLE
Update to .NET 8 and Remove .NET Core 3.1

### PR DESCRIPTION
**Pull request by aatchison14**
update target frameworks to remove .NET Core 3.1 and add .NET 8

### CODE REVIEW

File: Tests/RockLib.HealthChecks.Tests/HealthCheckResultTests.cs [hunk 1]
```diff
@@ -164,7 +164,7 @@ public void PropertySettersSetDictionaryValues()
             result["observedValue"].Should().Be("FakeObservedValue");
             result["observedUnit"].Should().Be("FakeObservedUnit");
             result["status"].Should().Be(HealthStatus.Warn);
-            result["affectedEndpoints"].Should().BeEquivalentTo(new[] { "FakeAffectedEndpoint" });
+            result["affectedEndpoints"].Should().BeEquivalentTo(new List<string> { "FakeAffectedEndpoint" });
```
[the remark this answers is not in the document]
Comment (NeilSchiefer) [reply Tests/RockLib.HealthChecks.Tests/HealthCheckResultTests.cs:167]: weird that the other one doesn't trigger it.

[the remark this answers is not in the document]
Comment (aatchison14) [reply Tests/RockLib.HealthChecks.Tests/HealthCheckResultTests.cs:167]: agreed

[the remark this answers is not in the document]
Comment (JasonBock) [reply Tests/RockLib.HealthChecks.Tests/HealthCheckResultTests.cs:167]: I don't think that's Rider-specific; more that it's coming from .NET analyzers. I'd personally change it to using collection expressions. 

"When to suppress warnings - You're not concerned about the performance impact of creating a constant array for each invocation." I think this is a case for that :). Just put a `<NoWarn>` in the .csproj file

**Review by JasonBock — APPROVED** [review]

**Comment by NeilSchiefer** [comment]
Looks like the tests are failing now.

**Review by NeilSchiefer — APPROVED** [review]

**Pull request merged**